control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -292,37 +292,7 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, broker *eventing.Broker)
 func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker) reconciler.Event {
 	logger := kafkalogging.CreateFinalizeMethodLogger(ctx, broker)
 
-	// Get contract config map.
-	contractConfigMap, err := r.GetOrCreateDataPlaneConfigMap(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to get contract config map %s: %w", r.DataPlaneConfigMapAsString(), err)
-	}
-
-	logger.Debug("Got contract config map")
-
-	// Get contract data.
-	ct, err := r.GetDataPlaneConfigMapData(logger, contractConfigMap)
-	if err != nil {
-		return fmt.Errorf("failed to get contract: %w", err)
-	}
-
-	logger.Debug("Got contract data from config map", zap.Any(base.ContractLogKey, ct))
-
-	if err := r.DeleteResource(ctx, logger, broker.GetUID(), ct, contractConfigMap); err != nil {
-		return err
-	}
-
-	// We update receiver and dispatcher pods annotation regardless of our contract changed or not due to the fact
-	// that in a previous reconciliation we might have failed to update one of our data plane pod annotation, so we want
-	// to update anyway remaining annotations with the contract generation that was saved in the CM.
-	// Note: if there aren't changes to be done at the pod annotation level, we just skip the update.
-
-	// Update volume generation annotation of receiver pods
-	if err := r.UpdateReceiverPodsAnnotation(ctx, logger, ct.Generation); err != nil {
-		return err
-	}
-	// Update volume generation annotation of dispatcher pods
-	if err := r.UpdateDispatcherPodsAnnotation(ctx, logger, ct.Generation); err != nil {
+	if err := r.deleteResourceFromContractConfigMap(ctx, logger, broker); err != nil {
```

Comment:
note: this function extraction is not a refactoring but it allows us to continue finalization while skipping ConfigMap changes as the comment on `apierrors.IsForbidden(err)` describes